scylla-cql/src/frame/request/auth_response.rs

```diff
@@ -1,37 +1,25 @@
 use crate::frame::frame_errors::ParseError;
 use bytes::BufMut;
-use std::convert::TryInto;
 
 use crate::frame::request::{Request, RequestOpcode};
-use crate::frame::Authenticator;
 
 // Implements Authenticate Response
 pub struct AuthResponse {
-    pub username: Option<String>,
-    pub password: Option<String>,
-    pub authenticator: Authenticator,
+    pub response: Option<Vec<u8>>,
 }
 
 impl Request for AuthResponse {
     const OPCODE: RequestOpcode = RequestOpcode::AuthResponse;
 
     fn serialize(&self, buf: &mut impl BufMut) -> Result<(), ParseError> {
-        if self.username.is_none() || self.password.is_none() {
-            return Err(ParseError::BadDataToSerialize(
-                "Bad credentials: username or password missing. You can use SessionBuilder::user(\"user\", \"pass\") to provide credentials.".to_string(),
-            ));
-        }
-        let username_as_bytes = self.username.as_ref().unwrap().as_bytes();
-        let password_as_bytes = self.password.as_ref().unwrap().as_bytes();
+        if let Some(bytes) = self.response.clone() {
```

Comment:
Please be mindful of clones and avoid them if you can. Here, you probably can do `self.response.as_ref()` instead of `self.response.clone()`.